pkg/core/resources.go

```diff
@@ -18,15 +18,19 @@ type (
 	// Construct describes a resource at the source code, Klotho annotation level
 	Construct interface {
 		BaseConstruct
-		// Provenance returns the AnnotationKey that the construct was created by
-		Provenance() AnnotationKey
+
+		// AnnotationCapability returns the annotation capability of the construct. This helps us tie the annotation types to the constructs for the time being
+		AnnotationCapability() string
 	}
 
+	// BaseConstructsRefSet is a set of BaseConstructs
+	BaseConstructSet map[BaseConstruct]struct{}
+
 	// Resource describes a resource at the provider, infrastructure level
 	Resource interface {
 		BaseConstruct
-		// KlothoConstructRef returns AnnotationKey of the klotho resource the cloud resource is correlated to
-		KlothoConstructRef() AnnotationKeySet
```

Comment:
Seems like AnnotationKeySet can be removed?

Reply:
yeah i didnt know if we were going to want to keep it for the compiler side, which is why i left it in, but i could remove it if we want